### PR TITLE
chore: Include stepConfiguration and subStepConfiguration properties in start events

### PR DESCRIPTION
**Pull request by connorlanigan**
### Description

Previously, we've included the `stepConfiguration` and `subStepConfiguration` properties in the `funnelChange` and `funnelStepChange` events (https://github.com/cloudscape-design/components/pull/1509, https://github.com/cloudscape-design/components/pull/1511).

With this PR, we also include these properties in the corresponding start event, so that they can be used as a reference when the change events are emitted (otherwise it is difficult to figure out what changed).

Related links, issue #, if available: n/a

### How has this been tested?
Extended unit tests

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/wizard/__tests__/wizard.test.tsx [hunk 1]
```diff
@@ -407,7 +407,7 @@ test('raises a warning when setting activeStepIndex without onNavigate listener'
 test('does not perform navigation when used in controlled mode', () => {
   const [wrapper] = renderDefaultWizard({ activeStepIndex: 1, onNavigate: () => {} });
   const checkStep = () => {
-    expect(wrapper.findContent()!.getElement()).toHaveTextContent(DEFAULT_STEPS[1].content as string);
+    expect(wrapper.findContent()!.getElement()).toHaveTextContent('Step 2, substep oneStep 2, substep two');
```
Comment (pan-kot) [inline src/wizard/__tests__/wizard.test.tsx:410]: What this change is for? The content of the updated DEFAULT_STEPS seems as still matching the output.

File: src/internal/analytics/components/analytics-funnel.tsx [hunk 2]
```diff
@@ -201,6 +206,19 @@ export const AnalyticsFunnelStep = (props: AnalyticsFunnelStepProps) => {
   return <InnerAnalyticsFunnelStep {...props} key={props.stepNumber} />;
 };
 
+function getSubStepConfiguration() {
+  const subSteps = Array.from(document.querySelectorAll<HTMLElement>(getSubStepAllSelector()));
+
+  const subStepConfiguration = subSteps.map((substep, index) => {
+    const name = substep.querySelector(getSubStepNameSelector())?.textContent ?? '';
```
Comment (pan-kot) [inline src/internal/analytics/components/analytics-funnel.tsx:213]: Should we include the sub-steps w/o a name at all?

Comment (pan-kot) [reply src/internal/analytics/components/analytics-funnel.tsx:213]: Is there a reason why here we use textContent and in getNameFromSelector we use innerText?

File: src/internal/analytics/__tests__/components/analytics-funnel.test.tsx [hunk 3]
```diff
@@ -332,10 +332,29 @@ describe('AnalyticsFunnelStep', () => {
     expect(FunnelMetrics.funnelStepStart).toHaveBeenCalledTimes(1);
     expect(FunnelMetrics.funnelStepStart).toHaveBeenCalledWith({
       funnelInteractionId: mockedFunnelInteractionId,
+      stepName: undefined,
```
Comment (pan-kot) [inline src/internal/analytics/__tests__/components/analytics-funnel.test.tsx:335]: Why stepName is undefined here? Is there a real difference between this value being an empty string and undefined?

File: src/wizard/internal.tsx [hunk 4]
```diff
@@ -225,3 +219,11 @@ function useFunnelChangeEvent(funnelInteractionId: string | undefined, steps: Wi
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [funnelInteractionId, stepTitles]);
 }
+
+export function getStepConfiguration(steps: WizardProps['steps']) {
```
[the remark this answers is not in the document]
Comment (connorlanigan) [reply src/wizard/internal.tsx:223]: Good idea

File: src/wizard/__tests__/wizard.test.tsx (hunk 1 above)
Comment (connorlanigan) [reply src/wizard/__tests__/wizard.test.tsx:410]: The DEFAULT_STEPS content is now JSX, which means that we can't directly compare it anymore with the text content that is rendered in the DOM.

File: src/internal/analytics/components/analytics-funnel.tsx (hunk 2 above)
Comment (connorlanigan) [reply src/internal/analytics/components/analytics-funnel.tsx:213]: Good catch, I don't think there was any particular reason. The advantage of `innerText` is that it doesn't include invisible text. I'll update this to use `innerText`.

Steps without a name should still be included. This only happens for containers that don't have a heading, which we still want to be recognisable in the event data.

File: src/internal/analytics/__tests__/components/analytics-funnel.test.tsx (hunk 3 above)
Comment (connorlanigan) [reply src/internal/analytics/__tests__/components/analytics-funnel.test.tsx:335]: I've updated the code in the next revision to have a proper step name

File: src/wizard/__tests__/wizard.test.tsx (hunk 1 above)
Comment (pan-kot) [reply src/wizard/__tests__/wizard.test.tsx:410]: got it 👍

File: src/internal/analytics/components/analytics-funnel.tsx (hunk 2 above)
Comment (pan-kot) [reply src/internal/analytics/components/analytics-funnel.tsx]: Should we have a better default than an empty string in that case? Like `name.trim() || null`?

**Review by pan-kot — APPROVED** [review]

**Review by pan-kot — APPROVED** [review]

**Pull request merged**